Source/monster.cpp

```diff
@@ -3745,7 +3745,7 @@ void monster_some_crypt()
 void InitGolems()
 {
 	if (!setlevel) {
-		for (int i = 0; i < MAX_PLRS; i++)
+		for (size_t i = 0; i < Players.size(); i++)

```

Comment:
Dis this break vanilla compatibility? 
With this change we only add one golem in single player.
That means the first normal monster could be at index 1.
But vanilla often checks for golems with ``if (monsterId < MAX_PLRS)``

Reply:
Vanilla also broke it self since it would spawn 8 golems for set levels :sweat_smile:

Reply:
Thanks to qndel this is fixed. 👍 #2822
But this vanilla bug is different.
It simply creates more golems at their holding cell.
That creates garbage but don't harm or?

Reply:
I'd keep the index at 4 like before, otherwise what happens when we load vanilla SP save in devx? golems 1-3 will be treated as monsters? What if you load devx save in vanilla? monsters 1-3 will be treatd as golems? :D

Reply:
Wouldn't loading vanilla SP save in DevX with 1-3 as monsters instead of golems not matter, because 1-3 would never be spawned golems given there's no player ever in slots 1-3 in single player?

That said, loading a DevX save with monsters 1-3 existing could maybe introduce interesting behavior in vanilla. xD

Reply:
Yeah let's revert this to `MAX_PLRS` just in case

Reply:
This is now `MAX_PLAYERS` again